packages/widgets/src/menu.ts

```diff
@@ -1485,10 +1556,11 @@ namespace Private {
     MessageLoop.sendMessage(menu, Widget.Msg.UpdateRequest);
 
     // Get the current position and size of the main viewport.
-    let px = window.pageXOffset;
-    let py = window.pageYOffset;
-    let cw = document.documentElement.clientWidth;
-    let ch = document.documentElement.clientHeight;
+    const windowData = getLatestWindowData() || _getWindowData();
```

Comment:
Should be moved one line up (prior to update request) as update can trigger DOM mods.